mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/ExtensionDevelopmentFramework.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclaration;
+import org.mule.runtime.extension.internal.property.DevelopmentFrameworkModelProperty;
+
+/**
+ * The main extension development frameworks.
+ *
+ * @since 1.6
+ */
+public enum ExtensionDevelopmentFramework {
+
+  JAVA_SDK, XML_SDK, MULE_SDK, MULE_DSL;
+
+  /**
+   * @param extensionModel The {@link ExtensionModel}.
+   * @return The {@link ExtensionDevelopmentFramework} used to develop the extension represented by given {@code extensionModel}.
+   *         Defaults to {@link ExtensionDevelopmentFramework#JAVA_SDK} if it cannot be determined.
+   */
+  public static ExtensionDevelopmentFramework getExtensionDevelopmentFramework(ExtensionModel extensionModel) {
```

Comment:
move this to an utils class, ideally outside this project. This enum is API. Should not be coupled to the ExtensionModel nor the ModelProperty.